pint/models/timing_model.py

```diff
@@ -774,14 +792,42 @@ def designmatrix(self, toas, acc_delay=None, scale_by_F0=True, \
             M[:, mask] /= F0.value
         return M, params, units, scale_by_F0
 
-    def read_parfile(self, filename):
-        """Read values from the specified parfile into the model parameters."""
+    def read_parfile(self, parfile):
+        """Read parameter values from a .parfile.
+
+           Parameter
+           ---------
+           parfile: str or .par file line lists
+               .par file name.
+        """
+        if isinstance(parfile, str):
+            pfile = open(parfile, 'r')
+            plines = pfile.readlines()
+        elif isinstance(parfile, (list, tuple)):
+            plines = parfile
+        else:
+            raise ValueError("Acceptable .par file format: `filename` and "
+                             "`.par file line list.`")
+        self.read_parfile_string(plines)
+        if isinstance(parfile, str):
+            pfile.close()
+
+    def read_parfile_string(self, par_strings):
```

Comment:
Is there any need for a separate function here rather than just making read_parfile accept both types?